code/game/area/areas.dm

```diff
@@ -170,6 +174,25 @@
 					spawn(0)
 						E.open()
 
+// Activate all retention fields!
+/area/proc/arfgs_activate()
+	if(!arfgs_active)
+		arfgs_active = TRUE
+		if(!all_arfgs)
+			return
+		for(var/obj/machinery/atmospheric_field_generator/E in all_arfgs)
+			E.generate_field() //don't need to check powered state like doors, the arfgs handles it on its end
+			E.wasactive = TRUE

```

Comment:
Should this get set every time `generate_field()` is called?

Reply:
`wasactive` is a slight hack that allows them to autoreboot if they're knocked out by an EMP or a low-power explosion. without it they stay offline.